### PR TITLE
Restrict available codecs in room for recording component

### PR DESCRIPTION
**Pull request by Karolk99**
## Acknowledging the stipulations set forth:
 - [x] I hereby confirm that a Pull Request involving updates to the Software Development Kit (SDK) has been smoothly merged, currently awaits processing, or is otherwise deemed unnecessary in this context.
 - [x] I also affirm that another Pull Request, specifically addressing updates to the documentation body (commonly referred to as 'docs'), has either been successfully incorporated, is in the process of review, or is considered superfluous under the prevailing circumstances.

### CODE REVIEW

File: lib/jellyfish/room.ex [hunk 1]
```diff
@@ -281,9 +281,9 @@ defmodule Jellyfish.Room do
         Logger.warning("Unable to add component: incompatible codec")
         {:reply, {:error, :incompatible_codec}, state}
 
-      {:error, :reached_components_limit_hls} ->
+      {:error, :reached_components_limit} ->
         Logger.warning("Unable to add component: reached components limit")
-        {:reply, {:error, :reached_components_limit_hls}, state}
+        {:reply, {:error, :reached_components_limit}, state}
```
Comment (Rados13) [inline lib/jellyfish/room.ex:286]: I think we should mention which component type limit we reached.

**Review by Rados13 — COMMENTED** [review]
Also one test failed.

File: lib/jellyfish/room.ex (hunk 1 above)
Comment (Karolk99) [reply lib/jellyfish/room.ex:286]: Don't you think that you should know based on the request you did

Comment (Karolk99) [reply lib/jellyfish/room.ex:286]: My mistake it's internal log

**Review by Rados13 — APPROVED** [review]

File: lib/jellyfish/room.ex [hunk 2]
```diff
@@ -789,16 +791,16 @@ defmodule Jellyfish.Room do
 
   defp check_component_allowed(RTSP, %{config: %{video_codec: video_codec}}) do
     # Right now, RTSP component can only publish H264, so there's no point adding it
-    # to a room which enforces another video codec, e.g. VP8
-    if video_codec in [:h264, nil],
```
Comment (sgfn) [inline lib/jellyfish/room.ex:793]: As discussed, please revert this change

**Review by sgfn — APPROVED** [review]
